File: OffByOne/src/OffByOne.LanguageCore/Ast/Expressions/Unary/NotExpression.cs
namespace OffByOne.LanguageCore.Ast.Expressions.Unary
{
    using OffByOne.LanguageCore.Ast.Expressions.Unary.Base;

    public class NotExpression : UnaryExpression
    {
        public NotExpression(Expression expression) 
            : base(expression)
        {
        }
    }
}

File: Loading/QL/src/ast/expression/NotExpression.java
package ast.expression;

import ast.ExpressionVisitor;
import ast.atom.Atom;
import ast.type.Type;
import semantic.Environment;

public class NotExpression extends UnaryExpression {

	public NotExpression(Expression lhs, int line) {
		super(lhs, line);
	}

	@Override
	public <T> T accept(ExpressionVisitor<T> v) {
		return v.visit(this);
	}
//	@Override
//	public Atom evaluate() {
//		return getLhs().evaluate().not();
//	}
}
